frontend/archive/src/main/java/org/pytorch/serve/archive/model/ModelArchive.java

```diff
@@ -55,7 +54,7 @@ public static ModelArchive downloadModel(
             throw new ModelNotFoundException("empty url");
         }
 
-        String marFileName = FilenameUtils.getName(url);
+        String marFileName = ArchiveUtils.getFilenameFromUrl(url);
```

Comment:
Yes, this is correct, the call path is [ModelArchive.downloadModel](https://github.com/pytorch/serve/blob/c2cdcfbd5b69580b96880b7df112d022cc15bdc7/frontend/archive/src/main/java/org/pytorch/serve/archive/model/ModelArchive.java#L46) -> [ArchiveUtils.downloadArchive](https://github.com/pytorch/serve/blob/c2cdcfbd5b69580b96880b7df112d022cc15bdc7/frontend/archive/src/main/java/org/pytorch/serve/archive/utils/ArchiveUtils.java#L93) -> [HttpUtils.copyURLToFile](https://github.com/pytorch/serve/blob/c2cdcfbd5b69580b96880b7df112d022cc15bdc7/frontend/archive/src/main/java/org/pytorch/serve/archive/model/s3/HttpUtils.java#L22) -> [FileUtils.copyURLToFile](https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/FileUtils.html#copyURLToFile(java.net.URL,%20java.io.File)).
`org.apache.commons.io.FileUtils.copyURLToFile` throws `IOException` because of the `destination` we pass to it in the case of S3 presigned URL.

For example, in the current implementation, in case of S3 pre-signed URL, we call `org.apache.commons.io.FileUtils.copyURLToFile` with the following arguments:

**source**: `https://test-account.s3.us-west-2.amazonaws.com/mar_files/resnet-18.mar?response-content-disposition=inline&X-Amz-Security-Token=token&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20230614T182131Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=credential&X-Amz-Signature=signature`

**destination**: `resnet-18.mar?response-content-disposition=inline&X-Amz-Security-Token=token&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20230614T182131Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=credential&X-Amz-Signature=signature`

and this fails with `IOException`.

For testing, I hardcoded the `destination` in the [HttpUtils.copyURLToFile](https://github.com/pytorch/serve/blob/c2cdcfbd5b69580b96880b7df112d022cc15bdc7/frontend/archive/src/main/java/org/pytorch/serve/archive/model/s3/HttpUtils.java#L22) to `resnet-18.mar` and the download succeeded.

Therefore, I've fixed the the implementation to correctly identify `filename` from model archive URL that contains additional parameters after the filename.

With the fix in this PR, the arguments are passed correctly:
**source**: `https://test-account.s3.us-west-2.amazonaws.com/mar_files/resnet-18.mar?response-content-disposition=inline&X-Amz-Security-Token=token&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20230614T182131Z&X-Amz-SignedHeaders=host&X-Amz-Expires=43200&X-Amz-Credential=credential&X-Amz-Signature=signature`

**destination**: `resnet-18.mar`